src/variant.h

```diff
@@ -149,12 +149,14 @@ struct Variant {
   bool flagPieceSafe = false;
   bool checkCounting = false;
   int connectN = 0;
+  PieceSet connectPieceTypes = CHESS_PIECES;
```

Comment:
I think we should either set the default to NO_PIECE_SET, so that no useless calculation is done for non-connect variants, or set it to all piece types and in that case adjust the logic to narrow it down to actually used piece types.

The latter option is what I tend to prefer in recent times, because instead of forcing the user to do consistent configuration, it tries to handle conflicting information gracefully. E.g., one could set the default to all piece types, but then in `Variant::conclude` & the connection piece types with the actually used piece types, and set it to no piece types if none of the connection options are used. This way all of the existing connection variants would still work without changing the INI and at the same time non-connection games would be handled cleanly.

Reply:
I did the second, start with all pieces, then optimize to either no pieces, or pieces actually in play.